provider/core/src/main/java/org/arquillian/pact/provider/core/PactProviderExtension.java

```diff
@@ -11,5 +16,10 @@ public void register(ExtensionBuilder builder) {
                 .observer(PactProviderConfigurator.class)
                 .observer(HttpTargetCreator.class)
                 .service(ResourceProvider.class, HttpTargetResourceProvider.class);
+
+        if(Validate.classExists("org.jboss.arquillian.container.test.spi.client.deployment.AuxiliaryArchiveAppender")) {
```

Comment:
This is a general thing, as you are using this approach in several different extensions - it's definitely not refactoring friendly and if we change class name or package structure it can go unnoticed. Maybe some logging to have at least a chance to see a WARNING might help?